src/probnum/filtsmooth/gaussfiltsmooth/iterated_kalman.py

```diff
@@ -94,34 +104,34 @@ def iterated_filtsmooth(self, dataset, times, **kwargs):
             posterior = self.smooth(posterior)
         return posterior
 
-    def predict(self, start, stop, randvar, linearise_at=None, **kwargs):
+    def predict(self, start, stop, randvar, linearise_at=None, intermediate_step=None):
         """(Possibly iterated) prediction step."""
-        pred_rv, info_pred = self.dynamod.transition_rv(
-            randvar, start, stop=stop, linearise_at=linearise_at, **kwargs
+        pred_rv, info_pred = self.dynamics_model.transition_rv(
+            randvar, start, stop=stop, linearise_at=linearise_at, step=intermediate_step
```

Comment:
I'm a bit confused about `step=intermediate_step`. I'm not really into the actual code, so I'm just wondering, but did you consider renaming "step" => "intermediate_step"? Or the other way round?

Reply:
I renamed it, because intermediate_step is something that is passed to `filter`, though not related at all to the mesh-width of the observations, and instead a quantity that is used in each prediction. Therefore I wanted to distinguish it from a `step` to avoid confusion. Does this make sense or is it unnecessary?